asset-transfer-basic/rest-api-typescript/README.md

```diff
@@ -0,0 +1,184 @@
+
+The primary aim of this sample is to show how to write a long running client application using the Fabric Node SDK
+
+The REST API is intended to work with the [basic asset transfer example](https://github.com/hyperledger/fabric-samples/tree/main/asset-transfer-basic)
+
+To install the basic asset transfer chaincode on a local Fabric network, follow the [Using the Fabric test network](https://hyperledger-fabric.readthedocs.io/en/release-2.2/test_network.html) tutorial
+
+## Overview
+
+The sample creates two long lived connections to a Fabric network in order to submit and evaluate transactions using two different identities
+
+To ensure requests respond quickly enough to avoid timeouts, all submit transactions are queued for processing and will be retried if they fail
+
+Submit transactions are retried if they fail with any error, except for errors from the smart contract, or duplicate transaction errors
+
+Alternatively you might prefer to modify the sample to only retry transactions which fail with specific errors instead, for example:
+- MVCC_READ_CONFLICT
+- PHANTOM_READ_CONFLICT
+- ENDORSEMENT_POLICY_FAILURE
+- CHAINCODE_VERSION_CONFLICT
+- EXPIRED_CHAINCODE
+
+## Usage
+
+**Note:** these instructions should work with the release-2.2 branch of `fabric-samples` but later versions require some changes
+
+To build and start the sample REST server, you'll need to [download and install an LTS version of node](https://nodejs.org/en/download/)
+
+Clone this repository and change to the `fabric-rest-sample/asset-transfer-basic/rest-api-typescript` directory before running the following commands
+
+Install dependencies
+
+```shell
+npm install
+```
+
+Build the REST server
+
+```shell
+npm run build
+```
+
+Create a `.env` file to configure the server for the test network (make sure TEST_NETWORK_HOME is set to the fully qualified `test-network` directory)
+
+```shell
+TEST_NETWORK_HOME=$HOME/fabric-samples/test-network npm run generateEnv
+```
+
+Start a Redis server
+
+```shell
+npm run start:redis
+```
+
+Start the sample REST server
+
+```shell
+npm run start:dev
+```
+
+### Docker image
+
+Alternatively, run the following commands in the `fabric-rest-sample/asset-transfer-basic/rest-api-typescript` directory to start the sample in a Docker container
+
+Build the Docker image
+
+```shell
+docker build -t fabric-rest-sample .
+```
+
+Create a `.env` file to configure the server for the test network (make sure `TEST_NETWORK_HOME` is set to the fully qualified `test-network` directory and `AS_LOCAL_HOST` is set to `false` so that the server works inside the Docker Compose network)
+
+```shell
+TEST_NETWORK_HOME=$HOME/fabric-samples/test-network AS_LOCAL_HOST=false npm run generateEnv
+```
+
+Start the sample REST server and Redis server
+
+```shell
+docker-compose up -d
+```
+
+## REST API
+
+If everything went well, you can now open a new terminal and try out some basic asset transfer REST calls!
```

Comment:
Should the `ready` and the `live` APIs be mentioned here...

Reply:
I'm leaning towards keeping it simple and not mentioning them in the readme. Seems like too much of a tangent into hosting for a high level run through.